src/main/java/com/zendesk/maxwell/bootstrap/AsynchronousBootstrapper.java

```diff
@@ -0,0 +1,162 @@
+package com.zendesk.maxwell.bootstrap;
+
+import com.zendesk.maxwell.*;
+import com.zendesk.maxwell.producer.AbstractProducer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.LinkedList;
+import java.util.Queue;
+
+public class AsynchronousBootstrapper extends AbstractBootstrapper {
+
+	static final Logger LOGGER = LoggerFactory.getLogger(MaxwellReplicator.class);
+
+	private Thread thread = null;
+	private Queue<RowMap > queue = new LinkedList<>();
+	private RowMap bootstrappedRow = null;
+	private RowMapBufferByTable skippedRows = null;
+	private SynchronousBootstrapper synchronousBootstrapper = getSynchronousBootstrapper();
+
+	public AsynchronousBootstrapper( MaxwellContext context ) throws IOException {
+		super(context);
+		skippedRows = new RowMapBufferByTable();
+	}
+
+	@Override
+	public boolean isStartBootstrapRow(RowMap row) {
+		return synchronousBootstrapper.isStartBootstrapRow(row);
+	}
+
+	@Override
+	public boolean isCompleteBootstrapRow(RowMap row) {
+		return synchronousBootstrapper.isCompleteBootstrapRow(row);
+	}
+
+	@Override
+	public boolean isBootstrapRow(RowMap row) {
+		return synchronousBootstrapper.isBootstrapRow(row);
+	}
```

Comment:
since these functions are side-effect free and there's only one implementation (in synchronousBootstrapper), let's move them to the abstract class.